notebooks/nbconverted/VAE_test.py

```diff
@@ -0,0 +1,338 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Image-based profiles VAEs
+
+# In[1]:
+
+
+import numpy as np
+import pandas as pd
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from cytotraj.utils.model_utils import load_data
+from torch.utils.data import Dataset
+
+# ## Building VAE class for Image-based profiles and Loss Function
+#
+
+# In[9]:
+
+
+class ImageProfileVAE(nn.Module):
+    """
+    VAE architecture for Image-based profiles
+    """
+
+    def __init__(self, input_dim, latent_dim):
+        """Initialize the Variational Autoencoder (VAE).
+
+        Parameters:
+        ----------
+        input_dim : int
+            The number of input features (dimensions) in the data.
+        latent_dim : int
+            The dimension of the latent space.
+
+        Returns:
+        -------
+        None
+        """
+        super(ImageProfileVAE, self).__init__()
+
+        # building the enconder sequence
+        self.encoder = nn.Sequential(
+            nn.Linear(input_dim, 64), nn.ReLU(), nn.Linear(64, 32), nn.ReLU()
+        )
+
+        # define the layers responsible for mapping the output of the encoder neural network
+        self.fc_mean = nn.Linear(32, latent_dim)
+        self.fc_logvar = nn.Linear(32, latent_dim)
+
+        # building the decoder sequence
+        self.decoder = nn.Sequential(
+            nn.Linear(latent_dim, 32),
+            nn.ReLU(),
+            nn.Linear(32, 64),
+            nn.ReLU(),
+            nn.Linear(64, input_dim),
+            nn.Sigmoid(),
```

Comment:
I would probably change this to a  linear activation function as the output instead. What is the range of your features?

Reply:
What is the range? I am assuming you are asking how many features are within the dataset. probably between 1000~2000 features, since the features were extracted by using CellProfiler